### PR TITLE
No need for a counter in hold_sync

### PR DESCRIPTION
**Pull request by SylvainCorlay**
It is simpler to keep a `_holding_sync` boolean than counting the number of nested context managers.

### CODE REVIEW

File: ipywidgets/widgets/widget.py [hunk 1]
```diff
@@ -380,12 +380,14 @@ def hold_sync(self):
         """Hold syncing any state until the context manager is released"""
         # We increment a value so that this can be nested.  Syncing will happen when
```
Comment (jasongrout) [inline ipywidgets/widgets/widget.py]: This looks like a smart way to do this.  This comment should be updated, though.

File: ipywidgets/widgets/widget.py [hunk 2]
```diff
@@ -378,14 +378,14 @@ def _allow_write(self):
     @contextmanager
     def hold_sync(self):
         """Hold syncing any state until the context manager is released"""
```
Comment (jasongrout) [inline ipywidgets/widgets/widget.py]: Can we change this description to "until the outermost context manager exits" to give some clue to the code below and more precisely document what is going on?

**Comment by jasongrout** [comment]
This looks good to me if you change the docs just slightly, as noted above.

**Comment by SylvainCorlay** [comment]
@jasongrout done.

**Comment by jasongrout** [comment]
Looks good.  Thanks!

**Pull request merged**